packages/backend/src/queryBuilder.ts

```diff
@@ -357,24 +358,39 @@ export const buildQuery = ({
     const sqlOrderBy =
         fieldOrders.length > 0 ? `ORDER BY ${fieldOrders.join(', ')}` : '';
 
-    const whereFilters = getFilterRulesFromGroup(filters.dimensions).map(
-        (filter) => {
-            const field = getFields(explore).find(
-                (d) => fieldId(d) === filter.target.fieldId,
+    const sqlFilterRule = (filter: FilterRule) => {
+        const field = getFields(explore).find(
+            (d) => fieldId(d) === filter.target.fieldId,
+        );
+        if (!field) {
+            throw new Error(
+                `Filter has a reference to an unknown dimension: ${filter.target.fieldId}`,
             );
-            if (!field) {
-                throw new Error(
-                    `Filter has a reference to an unknown dimension: ${filter.target.fieldId}`,
-                );
-            }
-            return renderFilterRuleSql(filter, field, q);
-        },
-    );
+        }
+        return renderFilterRuleSql(filter, field, q);
+    };
+
+    const getNestedFilterSQLFromGroup = (
```

Comment:
Really nice 👌 
We could use this for the metric filters as well since we are here.

Reply:
I updated the method to support metrics and applied it, tests pass, I'm not sure how to test this on the FE...